Panic in create cluster with "privateCluster.additionalEndpointServices.cloudformation"

**What happened?**
A description of actual behavior (with error messages).
I have specified a YAML file to create cluster with private cluster enabled
privateCluster:
  enabled: true
  additionalEndpointServices:
  - "autoscaling"
  - "logs"
  - "cloudformation"

**What you expected to happen?**
A clear and concise description of what the bug is.
I expect the process to go smoothly and create the cluster as desired.  However, this does not happen. It would appear that creating an additional enpoint service for cloudformation creates the runtime error as below:

```
[ℹ]  eksctl version 0.25.0
[ℹ]  using region eu-west-2
[✔]  using existing VPC (vpc-<redacted>) and subnets (private:[subnet-<redacted>subnet-<redacted>subnet-<redacted>] public:[])
[!]  custom VPC/subnets will be used; if resulting cluster doesn't function as expected, make sure to review the configuration of VPC/subnets
[ℹ]  nodegroup "nodegroup" will use "ami-<redacted>" [AmazonLinux2/1.16]
[ℹ]  using EC2 key pair "<redacted>"
[ℹ]  using Kubernetes version 1.16
[ℹ]  creating EKS cluster "<redacted>" in "eu-west-2" region with un-managed nodes
[ℹ]  1 nodegroup (nodegroup) was included (based on the include/exclude rules)
[ℹ]  will create a CloudFormation stack for cluster itself and 1 nodegroup stack(s)
[ℹ]  will create a CloudFormation stack for cluster itself and 0 managed nodegroup stack(s)
[ℹ]  if you encounter any issues, check CloudFormation console or try 'eksctl utils describe-stacks --region=eu-west-2 --cluster=<redacted>'
[ℹ]  Kubernetes API endpoint access will use provided values {publicAccess=true, privateAccess=true} for cluster "<redacted>" in "eu-west-2"
[ℹ]  2 sequential tasks: { create cluster control plane "<redacted>", 2 sequential sub-tasks: { 3 sequential sub-tasks: { tag cluster, update CloudWatch logging configuration, update cluster VPC endpoint access configuration }, create nodegroup "nodegroup" } }
[ℹ]  building cluster stack "eksctl-<redacted>-cluster"
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0x187f9ad]

goroutine 14 [running]:
github.com/weaveworks/eksctl/pkg/cfn/builder.importRouteTables(0x3903400, 0xc0001075c0, 0xc0000dce40, 0xc0006e4220, 0x0, 0x0)
        github.com/weaveworks/eksctl/pkg/cfn/builder/vpc.go:322 +0x46d
github.com/weaveworks/eksctl/pkg/cfn/builder.(*VPCResourceSet).importResources(0xc000656f00, 0x0, 0x0)
        github.com/weaveworks/eksctl/pkg/cfn/builder/vpc.go:264 +0x1be
github.com/weaveworks/eksctl/pkg/cfn/builder.(*VPCResourceSet).AddResources(0xc000656f00, 0x0, 0x0, 0xc000af2d98)
        github.com/weaveworks/eksctl/pkg/cfn/builder/vpc.go:101 +0x5bf
github.com/weaveworks/eksctl/pkg/cfn/builder.(*ClusterResourceSet).AddAllResources(0xc0001c3040, 0xc0001892c0, 0xc000221500)
        github.com/weaveworks/eksctl/pkg/cfn/builder/cluster.go:63 +0x6f
github.com/weaveworks/eksctl/pkg/cfn/manager.(*StackCollection).createClusterTask(0xc000656c00, 0xc00065e120, 0x2896c01, 0x2b4ccc0, 0xc0007bf370)
        github.com/weaveworks/eksctl/pkg/cfn/manager/cluster.go:34 +0x13d
github.com/weaveworks/eksctl/pkg/cfn/manager.(*createClusterTask).Do(0xc0007a77a0, 0xc00065e120, 0xc00065e120, 0x1)
        github.com/weaveworks/eksctl/pkg/cfn/manager/tasks.go:154 +0x41
github.com/weaveworks/eksctl/pkg/cfn/manager.doSingleTask(0xc00065e0c0, 0x3854f80, 0xc0007a77a0, 0x0)
        github.com/weaveworks/eksctl/pkg/cfn/manager/tasks.go:265 +0xec
github.com/weaveworks/eksctl/pkg/cfn/manager.doSequentialTasks(0xc00065e0c0, 0xc0007a7840, 0x2, 0x2)
        github.com/weaveworks/eksctl/pkg/cfn/manager/tasks.go:295 +0x70
created by github.com/weaveworks/eksctl/pkg/cfn/manager.(*TaskTree).DoAllSync
        github.com/weaveworks/eksctl/pkg/cfn/manager/tasks.go:118 +0x237
```

**How to reproduce it?**
Include the steps to reproduce the bug

below is my full YAML config file that created the runtime error.
(Please note that the YAML may not have formatted properly)
```
---
apiVersion: eksctl.io/v1alpha5
kind: ClusterConfig
metadata:
  name: <redacted>
  version: "1.16"
  region: eu-west-2
  tags:
   Project: K8sCluster
   Service: Calder
   Environment: Development

privateCluster:
  enabled: true
  additionalEndpointServices:
  - "autoscaling"
  - "logs"
  - "cloudformation"

vpc:
  id: "vpc-<redacted>"  # (optional, must match VPC ID used for each subnet below)
  #cidr: "10.42.0.0/16"       # (optional, must match CIDR used by the given VPC)
  autoAllocateIPv6: false
  subnets:
    private:
      eu-west-2a:
        id: "subnet-<redacted>"
      eu-west-2b:
        id: "subnet-<redacted>"
      eu-west-2c:
        id: "subnet-<redacted>"
   nodeGroups:
  - name: nodegroup
    labels:
      applicationType: Development
      Project: <redacted>
      nodegroup-type: frontend-workloads
    instanceType: t3a.medium
    desiredCapacity: 3
    volumeSize: 40
    volumeType: gp2
    ami: ami-<redacted>
    privateNetworking: true
    #amiFamily: AmazonLinux2
    ssh:
      allow: true
      publicKeyName: <redacted>
    iam:
      withAddonPolicies:
        autoScaler: true
        albIngress: true
        externalDNS: true
        certManager: true
        imageBuilder: true
cloudWatch:
  clusterLogging:
    enableTypes: ["audit", "authenticator", "api", "controllerManager", "scheduler"]
secretsEncryption:
  keyARN: "arn:aws:kms:eu-west-2:<redacted>"
```

**Anything else we need to know?**
What OS are you using, are you using a downloaded binary or did you compile eksctl, what type of AWS credentials are you using (i.e. default/named profile, MFA) - please don't include actual credentials though!

I updated my golang and kernel after the first crash thinking it might have to do with the underlying support in the system. But obviously it was not
Underlying system support are
go version go1.14.6 linux/amd64
kernel:  3.10.0-1127.18.2.el7.x86_64
Centos

**Versions**
Please paste in the output of these commands:
```
$ eksctl version
eksctl version
0.25.0

$ kubectl version
1.16
```

**Logs**
Include the output of the command line when running eksctl. If possible, eksctl should be run with debug logs. For example:
`eksctl get clusters -v 4`
Make sure you redact any sensitive information before posting.
If the output is long, please consider a Gist.

--- a/pkg/cfn/builder/vpc.go
+++ b/pkg/cfn/builder/vpc.go
@@ -319,6 +319,9 @@ func importRouteTables(ec2API ec2iface.EC2API, subnets map[string]api.Network) (
 	subnetRoutes := make(map[string]string)
 	for _, rt := range routeTables {
 		for _, rta := range rt.Associations {
+			if rta.Main != nil && *rta.Main {
+				return nil, errors.New("subnets must be associated with a non-main route table; eksctl does not modify the main route table")
+			}
 			subnetRoutes[*rta.SubnetId] = *rt.RouteTableId
 		}
 	}

--- a/pkg/cfn/builder/vpc_endpoint_test.go
+++ b/pkg/cfn/builder/vpc_endpoint_test.go
@@ -24,6 +24,7 @@ type vpcResourceSetCase struct {
 	clusterConfig  *api.ClusterConfig
 	expectedFile   string
 	createProvider func() api.ClusterProvider
+	err            string
 }
 
 var _ = Describe("VPC Endpoint Builder", func() {
@@ -48,8 +49,13 @@ var _ = Describe("VPC Endpoint Builder", func() {
 		rs := newResourceSet()
 		vpcResourceSet := NewVPCResourceSet(rs, vc.clusterConfig, provider)
 		vpcResource, err := vpcResourceSet.AddResources()
-		Expect(err).ToNot(HaveOccurred())
+		if vc.err != "" {
+			Expect(err).To(HaveOccurred())
+			Expect(err.Error()).To(ContainSubstring("subnets must be associated with a non-main route table"))
+			return
+		}
 
+		Expect(err).ToNot(HaveOccurred())
 		if vc.clusterConfig.PrivateCluster.Enabled {
 			vpcEndpointResourceSet := NewVPCEndpointResourceSet(provider, rs, vc.clusterConfig, vpcResource.VPC, vpcResource.SubnetDetails.Private, gfnt.NewString("sg-test"))
 			Expect(vpcEndpointResourceSet.AddResources()).To(Succeed())
@@ -167,6 +173,54 @@ var _ = Describe("VPC Endpoint Builder", func() {
 				return provider
 			},
 			expectedFile: "custom_vpc_private_endpoint_same_route_table.json",
+		}),
+		Entry("Private cluster with a user-supplied VPC having subnets with an explicit main route table association", vpcResourceSetCase{
+			clusterConfig: &api.ClusterConfig{
+				PrivateCluster: &api.PrivateCluster{
+					Enabled: true,
+				},
+				VPC: &api.ClusterVPC{
+					Network: api.Network{
+						ID: "vpc-custom",
+					},
+					Subnets: &api.ClusterSubnets{
+						Private: map[string]api.Network{
+							"us-west-2a": {
+								ID: "subnet-custom1",
+							},
+						},
+					},
+				},
+			},
+			createProvider: func() api.ClusterProvider {
+				provider := mockprovider.NewMockProvider()
+				output := &ec2.DescribeRouteTablesOutput{
+					RouteTables: []*ec2.RouteTable{
+						{
+							VpcId:        aws.String("vpc-custom"),
+							RouteTableId: aws.String("rt-main"),
+							Associations: []*ec2.RouteTableAssociation{
+								{
+									RouteTableId:            aws.String("rt-main"),
+									RouteTableAssociationId: aws.String("rtbassoc-custom1"),
+									Main:                    aws.Bool(true),
+								},
+								{
+									RouteTableId:            aws.String("rt-main"),
+									SubnetId:                aws.String("subnet-custom1"),
+									RouteTableAssociationId: aws.String("rtbassoc-custom2"),
+									Main:                    aws.Bool(false),
+								},
+							},
+						},
+					},
+				}
+				provider.MockEC2().On("DescribeRouteTables", mock.MatchedBy(func(input *ec2.DescribeRouteTablesInput) bool {
+					return len(input.Filters) > 0
+				})).Return(output, nil)
+				return provider
+			},
+			err: "subnets must be associated with a non-main route table",
 		}),
 	)
 })

